atlasdb-tests-shared/src/main/java/com/palantir/test/utils/EnvironmentVariablesExtension.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.test.utils;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+/** Assumes the given key will be set once. */
+public class EnvironmentVariablesExtension implements AfterEachCallback {
+
+    private final Map<String, String> testValues = new HashMap<>();
+    private final Map<String, String> originalValues = new HashMap<>();
+
+    public void set(String key, String value) {
+        testValues.put(key, value);
+        if (System.getenv().containsKey(key)) {
+            originalValues.put(key, System.getenv(key));
+        }
+
+        if (value == null) {
```

Comment:
I find it confusing the fact that we are using an API were passing `null` means removing. I'd rather just have us call the `setEnvironmentVariable` or `removeEnvironmentVariable` directly from the client code. And leave the `BeforeEach` and `AfterEach` to do the proper clean up and restoring of the variables, instead of trying to do this on each `set`